### PR TITLE
Phone validation

### LINKED ISSUE #2189: Phone number in contact info page should be validated
## Summary

Phone number in contact info page should be validated. Currently, it accepts the letters and less than 10 characters (full and fyi reports)

## Steps to reproduce

> How exactly can the bug be reproduced? Be very specific.

## Unresolved questions

> Are there any related issues you consider out of scope for this issue that could be addressed in the future?

### PR DESCRIPTION
**Pull request by ying-chen-cds**
Fixes (#Issue) 

# Description

> Add a loose validation (only allow numbers, space, brackets and hyphen) for phone number and remove the phrase "Enter a phone number that contains 10 digits."

# Any new packages installed?

> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ ] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

File: f2/src/forms/ContactInfoForm.js [hunk 1]
```diff
@@ -34,7 +37,7 @@ export const validate = (values) => {
     errors.email = 'contactinfoForm.email.warning'
   }
 
-  if (!values.phone) {
+  if (!new RegExp(phoneRegex).test(values.phone)) {
```
Comment (justinr86) [inline f2/src/forms/ContactInfoForm.js:40]: This validation is only applied to the full report. If I submit FYI report I can still enter "a" as the phone number.

File: f2/src/forms/ContactInfoForm.js [hunk 2]
```diff
@@ -25,6 +25,9 @@ import { Link as ReactRouterLink } from 'react-router-dom'
 export const validate = (values) => {
   const errors = {}
 
+  //a loose validation for phone number(only allow numbers, space, brackets and hyphen)
+  const phoneRegex = /^[0-9()\-\s]*$/g
```
Comment (justinr86) [inline f2/src/forms/ContactInfoForm.js:29]: This does allow more flexibility, but I can enter one digit. The form says to enter a valid phone number, in my opinion 1 is just as invalid as T,.

File: f2/src/locales/fr.json [hunk 3]
```diff
@@ -122,7 +122,7 @@
   "contactInfoPage.victimDetail": "Détails sur la victime",
   "contactinfoForm.email.warning": "Entrez une adresse courriel valide.",
   "contactinfoForm.fullName.warning": "Entrez votre nom complet",
-  "contactinfoForm.phone.warning": "Entrez un numéro de téléphone de 10 chiffres.",
+  "contactinfoForm.phone.warning": "Entrez un numéro de téléphone valide.",
```
Comment (justinr86) [inline f2/src/locales/fr.json:125]: Where did this French translation come from?

**Comment by justinr86** [comment]
This PR directly conflicts with #2194 opened by David.

**Comment by dengcn** [comment]
I think Fix to 2185 already covered this phone validation, fix to 2185 is waiting for business decision on if email or/and phone are required by team leader

File: f2/src/forms/ContactInfoForm.js (hunk 2 above)
Comment (ying-chen-cds) [reply f2/src/forms/ContactInfoForm.js:29]: This is a loose validation to meet Hong's requirement "at least validated to accept the digit numbers only and remove phrase (Enter a phone number that contains 10 digits.)"

File: f2/src/locales/fr.json (hunk 3 above)
Comment (ying-chen-cds) [reply f2/src/locales/fr.json:125]: I translated it. Where should usually the French translation come from?

**Comment by ying-chen-cds** [comment]
> I think Fix to 2185 already covered this phone validation, fix to 2185 is waiting for business decision on if email or/and phone are required by team leader

It looks like the fix 2185 just brought back the old phone validation, which would not work with the international phone number

File: f2/src/forms/ContactInfoForm.js (hunk 1 above)
Comment (ying-chen-cds) [reply f2/src/forms/ContactInfoForm.js:40]: Yeah, you are right. I didn't check FYI flow, I'll check that.

**Comment by ying-chen-cds** [comment]
This is closed since the requirement is changed and need to be re-implemented.

**Pull request closed without merging**